File: stack/Stack.java
package stack;

import java.util.ArrayList;
import java.util.List;

public class Stack<T> {
    private List<T> stack = new ArrayList<T>();

    public void push(T value) {
        stack.add(value);
    }

    public T pop() {
        T value = stack.get(size() - 1);
        stack.remove(size() - 1);
        return value;
    }

    public int size() {
        return stack.size();
    }

    public T top() {
        return stack.get(size() - 1);
    }

    public T bottom() {
        return stack.get(0);
    }

    public boolean isEmpty() {
        return stack.isEmpty();
    }

    public static void main(String[] args) {
        var stack = new Stack<Integer>();

        for (int i = 0; i < 10; i++) {
            stack.push(i);
        }

        System.out.println(stack.bottom());

        while (!stack.isEmpty()) {
            if (stack.top() % 2 == 0) {
                System.out.println(stack.top());
            }
            stack.pop();
        }
    }
}

File: stack/stack.py
class Stack:
    def __init__(self):
        self.__stack = []
    
    def push(self, value):
        self.__stack.append(value)
    
    def pop(self):
        return self.__stack.pop()
    
    def size(self):
        return len(self.__stack)

    def is_empty(self):
        return self.size() == 0;

    def top(self):
        return self.__stack[-1]

    def bottom(self):
        return self.__stack[0]


if __name__ == '__main__':
    elements = [i for i in range(10)]
    stack = Stack()
    
    for element in elements:
        stack.push(element)
    
    print(stack.bottom())
    
    while not stack.is_empty():
        if stack.top() % 2 == 0:
            print(stack.top())
        stack.pop()
